pkg/operator/operator.go

```diff
@@ -241,76 +273,237 @@ func (k *Kontroller) withLeaderElection() error {
 func (k *Kontroller) process() {
 	glog.V(4).Info("Going through a loop cycle")
 
-	nodelist, err := k.nc.List(v1meta.ListOptions{})
+	// find nodes with the after-reboot=true label and check if all provided
+	// annotations are set. if all annotations are set to true then remove the
+	// after-reboot=true label and set reboot-ok=false, telling the agent that
+	// the reboot has completed.
+	glog.V(4).Info("Checking for existence of configured after-reboot annotations")
+	err := k.checkAfterReboot()
 	if err != nil {
-		glog.Infof("Failed listing nodes %v", err)
+		glog.Errorf("Failed to check after reboot: %v", err)
 		return
 	}
 
-	glog.V(6).Infof("Found nodes: %+v", nodelist.Items)
+	// find nodes which just rebooted but haven't run after-reboot checks.
+	// remove any old annotations and add the after-reboot=true label.
+	glog.V(4).Info("Marking rebooted nodes with after-reboot label")
```

Comment:
nit: Can we consistently say "Labeling ..." or "Annotating ..." in log messages?